viewing_party/party.py

```diff
@@ -1,23 +1,181 @@
+    return average_rating
 
+def get_most_watched_genre(user_data):
+    genre_count = {}
+    most_watched_genre = ""
+    if not user_data["watched"]:
+        return None
+
+    for movie in user_data["watched"]:
+        if movie["genre"] not in genre_count:
+            genre_count[movie["genre"]] = 1
+        else:
+            genre_count[movie["genre"]] += 1
+
+    most_watched_genre = max(genre_count, key=genre_count.get)
+    return most_watched_genre
 
-# -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
+def get_unique_watched(user_data):
+    user_watched = []
+    for movie in user_data["watched"]:
+        user_watched.append(movie["title"])
+
+    friends_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] not in friends_watched:
+                friends_watched.append(movie["title"])
+
+    user_watched_set = set(user_watched)
+    friends_watched_set = set(friends_watched)
+
+    unique_movies_watched_user_only = user_watched_set - friends_watched_set
+
+    unique_movies_watched_user_only_list = []
+    for movie in user_data["watched"]:
+        if movie["title"] in unique_movies_watched_user_only:
+            unique_movies_watched_user_only_list.append(movie)
+
+    return unique_movies_watched_user_only_list
+
+def get_friends_unique_watched(user_data):
+    user_watched = []
+    for movie in user_data["watched"]:
+        user_watched.append(movie["title"])
+
+    friends_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] not in friends_watched:
+                friends_watched.append(movie["title"])
+
+    user_watched_set = set(user_watched)
+    friends_watched_set = set(friends_watched)
+
+    unique_movies_watched_friends_only = friends_watched_set - user_watched_set
+
+    seen_titles = []
+    unique_movies_watched_friends_only_list = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] in unique_movies_watched_friends_only and movie["title"] not in seen_titles:
+                unique_movies_watched_friends_only_list.append(movie)
+                seen_titles.append(movie["title"])
+
+    return unique_movies_watched_friends_only_list
 
-        
-# -----------------------------------------
 # ------------- WAVE 4 --------------------
-# -----------------------------------------
+def get_available_recs(user_data):
+    user_watched = []
+    for movie in user_data["watched"]:
+        user_watched.append(movie["title"])
+
+    subscriptions = user_data["subscriptions"]
+
+    friends_watched_movies = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_watched_movies:
+                friends_watched_movies.append(movie)

```

Comment:
The first part of this function should be looking at all the movies that the user has not watched but their friends have... I wonder if we have already written a function that will do this for us? Remember that we are absolutely able to use our other functions that we have written to to accomplish parts of what we need to do!